src/main/scala/uncore/converters/Tilelink.scala

```diff
@@ -466,15 +482,33 @@ class TileLinkIONarrower(innerTLId: String, outerTLId: String)
     data = smallput_data,
     wmask = Some(smallput_wmask))(outerConfig)
 
+  val atomic_addr = iacq.full_addr()
+  val atomic_addr_beat = atomic_addr(outerBlockOffset - 1, outerByteAddrBits)
+  val atomic_addr_byte = atomic_addr(outerByteAddrBits - 1, 0)
+  val atomic_data_sel = atomic_addr(outerByteAddrBits + log2Up(factor) - 1, outerByteAddrBits)
+  val atomic_data_vec = Vec(Seq.tabulate(factor) {
+    i => iacq.data((i + 1) * outerDataBits - 1, i * outerDataBits)
+  })
+
+  val put_atomic_acquire = PutAtomic(
+    client_xact_id = iacq.client_xact_id,
+    addr_block = iacq.addr_block,
+    addr_beat = atomic_addr_beat,
+    addr_byte = atomic_addr_byte,
+    atomic_opcode = iacq.op_code(),
+    operand_size = iacq.op_size(),
+    data = atomic_data_vec(atomic_data_sel))
+
   val sending_put = Reg(init = Bool(false))
 
   val pass_valid = io.in.acquire.valid && !stretch
 
   io.out.acquire.bits := MuxCase(Wire(io.out.acquire.bits, init=iacq), Seq(
-    (sending_put, put_block_acquire),
-    (shrink, get_block_acquire),
-    (smallput, put_acquire),
-    (smallget, get_acquire)))
+    sending_put -> put_block_acquire,
+    shrink -> get_block_acquire,
```

Comment:
In the previous mux_case shrink was named wideget, consistency seems nice here. I think wideget seems like a better name but don't have a strong preference.